crates/wallet/src/wallet/signer.rs

```diff
@@ -548,21 +566,24 @@ fn sign_psbt_ecdsa(
     secret_key: &secp256k1::SecretKey,
     pubkey: PublicKey,
     psbt_input: &mut psbt::Input,
-    hash: impl bitcoin::hashes::Hash + bitcoin::secp256k1::ThirtyTwoByteHash,
-    hash_ty: EcdsaSighashType,
+    hash: impl bitcoin::hashes::Hash<Bytes = [u8; 32]>,
+    sighash_type: EcdsaSighashType,
     secp: &SecpCtx,
     allow_grinding: bool,
 ) {
-    let msg = &Message::from(hash);
-    let sig = if allow_grinding {
+    let msg = &Message::from_digest(hash.to_byte_array());
```

Comment:
We could also just pass a `&Message` to the function `sign_psbt_ecdsa`

Reply:
@ValuedMammal Wouldn't it add more complexity/verbosity ? 🤔

Reply:
I'm thinking this module might eventually look more like in #1424, but if this works for now then that's great too